plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestTrinoRestCatalog.java

```diff
@@ -200,4 +200,36 @@ public void testPrefix()
                 .as("should fail as the prefix dev is not implemented for the current endpoint")
                 .hasMessageContaining("Malformed request: No route for request: POST v1/dev/namespaces");
     }
+
+    @Test
+    public void testMultipartNamespace()
+    {
+        String dotSeparator = ".";
+        TrinoCatalog catalog = createTrinoRestCatalog(false, ImmutableMap.of("prefix", "dev"));
+
+        ImmutableList<String> namespacesList = ImmutableList.of(
+                "level-1" + randomNameSuffix(),
+                "level-1" + randomNameSuffix() + dotSeparator + "level-2" + randomNameSuffix(),
+                "level-1" + randomNameSuffix() + dotSeparator + "level-2" + randomNameSuffix() + dotSeparator + "level-3" + randomNameSuffix());
+
+        for (String namespace : namespacesList) {
+            catalog.createNamespace(SESSION, namespace, ImmutableMap.of(), new TrinoPrincipal(PrincipalType.USER, SESSION.getUser()));
+        }
+
+        try {
+            assertThat(catalog.listNamespaces(SESSION)).as("catalog.listNamespaces")
+                    .hasSize(namespacesList.size())
+                    .containsAll(namespacesList);
+        }
+        finally {
+            for (String namespace : namespacesList) {
+                try {
+                    catalog.dropNamespace(SESSION, namespace);
+                }
+                catch (Exception e) {
+                    LOG.warn("Failed to clean up namespace: %s", namespace);
+                }
```

Comment:
Use stream api. Remove catch clause as it shouldn't be required.

Reply:
sure, thanks

Reply:
done